### ISSUE TITLE
`github` shim doesn't launch on Windows

### ISSUE THREAD

**Issue by shiftkey** [desktop/desktop#2563]
### Description

I currently can't launch the GitHub shim in any sort of Windows shell 🤔 

### Version

**GitHub Desktop version:** 0.8.0 and 0.8.1-beta2

**OS version:** `Microsoft Windows [Version 10.0.15063]`

### Steps to Reproduce

1. Launch application.
1. Select a repository
1. In the **Repository** menu, select **Open in External Shell** (either Command Prompt or PowerShell are demonstrating the problem for me).
1. Close Desktop 
1. Run `github .` from the command prompt

**Expected behavior:** GitHub Desktop launches, focuses on current folder

**Actual behavior:** Nothing

**Reproduces how often:** 100%

**Comment by shiftkey** [comment]
After cleaning up the `.bat` files involved and not finding any issues I fired up `procmon` and found this interesting chain of messages:

<img width="1434" src="https://user-images.githubusercontent.com/359239/29761055-97d8cd74-8c0a-11e7-96cc-e28bf1d3f1a7.png">

To me this looks like it's trying to find a `start` command on the PATH and run that, rather than use the shell's `start` function on Windows.

Anyway, turns out this looks like it's my fault when I wanted to ensure we handled untrusted input correctly in 995056713ec0abd1c98c1d34b845ef4334b64a57 (cc @j-f1 as you spotted this earlier). Now to figure out how we _should_ do this on Windows.
